### ISSUE TITLE
DPR decode_best_spans include spans from title

### ISSUE THREAD

**Issue by vinicius-cleves** [huggingface/transformers#10375]
## Environment info

- `transformers` version: 4.1.1

### Who can help
@lhoestq, @LysandreJik 

## Information

I believe there is a bug on the following line
```python
passage_offset = sequence_ids.index(self.sep_token_id, 2) + 1  # second sep id
```
It is on file `src/transformers/models/dpr/tokenization_dpr.py`. Next some context to this line:
```python
class CustomDPRReaderTokenizerMixin:
    ...
    def decode_best_spans(...) -> List[DPRSpanPrediction]:
        ...
        for doc_id in sorted_docs:
            ...
            # assuming question & title information is at the beginning of the sequence
            passage_offset = sequence_ids.index(self.sep_token_id, 2) + 1  # second sep id
            ...
        return nbest_spans_predictions[:num_spans]
```

The comments make me think that `passage_offset` refers to the start of the passage, after question and title. I feel that the intent behind `sequence_ids.index(self.sep_token_id, 2) ` was to select the second position where `self.sep_token_id` appears, but this doesn't happen, as this is selecting the first ocurrence of sep_token_id starting on token number 2. 

I believe an easy fix would be:
```python
title_offset = sequence_ids.index(self.sep_token_id) + 1  # first sep id
passage_offset = sequence_ids.index(self.sep_token_id, title_offset) + 1  # second sep id
```

**Comment by lhoestq** [comment]
Yes that's totally true. Good catch !
Could you open a PR to fix that please ?

Maybe this could have affected the performance of the DPR Reader a bit, but probably not significantly though since the logits of the tokens in the title have very low values. The model was trained to return answers from the passage, not from the  title.
